### DISCUSSION TITLE
Map for Home Assistant (includes zones/rooms)

### DISCUSSION THREAD

**Discussion by ottelo9** [rand256/valetudo#359] [Q&A]
Hi there. I'm actually still using the original Valetudo on my Gen2 (s50) together with Home Assistant and the [Lovelace Valetudo Map Card](https://github.com/TheLastProject/lovelace-valetudo-map-card). Now I want to flash Valetudo RE on my robot and try to get a better Map for my Home Assistant (HA). The actually problem is, if I start a zone cleanup the map in HA dont show the zone. The map only shows the forbidden zones.

Map in HA (generated via Lovelace Valetudo Map Card, installed via HACS):
(![2021_01_02_13_21_20_Overview_Home_Assistant](https://user-images.githubusercontent.com/33122175/103457247-bad6ac80-4cfd-11eb-8a03-08901f950c0d.png)))

questions:
- How can I display the map in HA with Valetudo RE displaying zones/rooms actually cleanin
- Can I use Lovelace Valetudo Map Card with Valetudo RE?
Thanks. @

**Comment by pidator** [comment]
Afaik Lovelace Valetudo Map Card is intend to be used with Valetudo Vanilla (>v0.6). There's only a few support for older Valetudo versions (or forks -  if I'm getting it right) with "Legacy Mode". So your questions are related to the support of Valetudo RE within the Lovelace Card and this can't be fulfilled by Valetudo RE - from my point of view it's an feature request for the Lovelace Card and should be placed there!?

**Comment by ottelo9** [comment]
No, I want to know how to display the map in Home Assistant including my rooms or zones with VALETUDO RE?

**Reply by ottelo9** [reply]
No one?

**Comment by rand256** [comment]
For a long time there is [valetudo mapper](https://github.com/rand256/valetudo-mapper) that receives the compressed mapdata from the device via mqtt and generates the png image of the map, that can be either served via http or published back to mqtt.

It can also publishthe parsed mapdata in "legacy" format that origin valetudo used initially. I'm not aware of what it uses now though.

**Reply by ottelo9** [reply]
@rand256 Thanks for your answer. Can I see my actual zone or room if I start zoned or room cleanup on the map, generated via valetudo mapper?

**Reply by rand256** [reply]
Yes if you set `drawCurrentlyCleanedZones` and `drawCurrentlyCleanedBlocks` options to true.

**Reply by ottelo9** [reply]
Hi there. I flashed my Roborock S5 to the newest Valetudo RE. Now I want to add the map to my Hassio Home Assistant (on Rpi). Is there a ready to use hassio addon for that?
